sdk/core/azure-core/src/main/java/com/azure/core/util/serializer/FlatteningSerializer.java

```diff
@@ -198,14 +197,14 @@ private static void escapeMapKeys(Object value, ClientLogger logger) {
             return;
         }
 
-        int mod = value.getClass().getModifiers();
-        if (Modifier.isFinal(mod) || Modifier.isStatic(mod)) {
-            return;
-        }
-
         for (Field f : getAllDeclaredFields(value.getClass())) {
-            // Why is this setting accessible to true?
-            f.setAccessible(true);
+            // only drill into classes defined in azure sdk
+            if (value.getClass().getPackage().getName().contains("com.azure")) {
```

Comment:
I'm good with this change as it helps prevent deeply escaping into classes outside of our ownership, but I'd love to get the thoughts of @srnagar and @JonathanGiles .

Reply:
I guess I'd need more context here - what is the downside to not doing this? By default magic strings make me a little anxious, so I'd prefer to not have this here if possible.

Reply:
It seems like this check is added as an optimization to short circuit the recursion. Will this gain us any significant perf improvements? If not, it may not be necessary to over-optimize here to minimize custom logic specific to package names.

Reply:
It's both an bit of an optimization but more of a safe-guard to prevent attempting to flatten out classes that we don't have reflective access to such as Java built-in classes such as `Constructor`, etc (which was an exception hit before this was added)

Reply:
Yes, like [here](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1010037&view=logs&j=cc8b1317-ae45-5611-4652-6b3ff0c8aa54&t=eb4c291c-cc36-5a28-3a59-910c03bf3803&l=36953) wherein we were trying to access the `clazz` property on the `ExpandableStringEnum` class.